### PR TITLE
Csu merge

### PR DESCRIPTION
**Pull request by bsdjhb**
This merges the "sed hack" for CSU and some followups.

I'm still waiting for this version to build but will give CI a chance to outpace my desktop.

### CODE REVIEW

File: lib/csu/mips64c128/Makefile [hunk 1]
```diff
@@ -1,63 +1,35 @@
...
+CFLAGS+=	-I${.CURDIR} -I${.CURDIR:H}/common \
+		-I${.CURDIR:H}/common-cheri \
+		-I${SRCTOP}/lib/libc/include
+CFLAGS+=	-mxgot -mllvm -mxmxgot
+CFLAGS+=	-DCRT_IRELOC_SUPPRESS
+# Purecap code requires crt_init_globals to be called before using jump tables
+CFLAGS+=	-fno-jump-tables
+
+FILES=		${OBJS}
+FILESMODE=      ${LIBMODE}
```
Comment (brooksdavis) [inline lib/csu/mips64c128/Makefile:15]: it might be with fixing the use of spaces in this area

Comment (bsdjhb) [reply lib/csu/mips64c128/Makefile:15]: It's all going to go away when we merge in the commit that moves duplicated lines to ../Makefile.inc.  I do need to get CI green though.

File: lib/csu/aarch64/Makefile [hunk 2]
```diff
@@ -17,31 +17,21 @@ FILESDIR=	${LIBDIR}
...
 
-crt1.s: crt1.c
-	${CC} ${CFLAGS} -S -o ${.TARGET} ${.CURDIR}/crt1.c
-	sed ${SED_FIX_NOTE} ${.TARGET}
+gcrt1.o: gcrt1_c.o crt1_s.o
+	${LD} ${_LDFLAGS} -o gcrt1.o -r crt1_s.o gcrt1_c.o
 
-crt1.o: crt1.s
-	${CC} ${CFLAGS:N-g} ${ACFLAGS} -c -o ${.TARGET} crt1.s
+crt1.o:	crt1_c.o crt1_s.o
```
Comment (jrtc27) [inline lib/csu/aarch64/Makefile:28]: Is this really upstream? :/

**Review by brooksdavis — APPROVED** [review]

File: lib/csu/common/crtend.c [hunk 3]
```diff
@@ -28,9 +28,11 @@ __FBSDID("$FreeBSD$");
 
 typedef void (*crt_func)(void);
 
-static crt_func __JCR_LIST__[] __section(".jcr") __used = {
+#ifndef __CHERI_PURE_CAPABILITY__
+static crt_func __JCR_END__[] __section(".jcr") __used = {
```
Comment (jrtc27) [inline lib/csu/common/crtend.c:32]: Has the name been fixed upstream?

File: sys/sys/elf_common.h [hunk 4]
```diff
@@ -745,6 +745,7 @@ typedef struct {
 #define	DT_MIPS_CHERI_CAPTABLE_MAPPING	0x7000c005 /* start of .captable_mapping */
 #define	DT_MIPS_CHERI_CAPTABLE_MAPPINGSZ	0x7000c006 /* size of .captable_mapping */
 
+#ifndef LOCORE
 enum MipsCheriFlags {
```
Comment (jrtc27) [inline sys/sys/elf_common.h:749]: We should probably make this macros...

File: lib/csu/aarch64/Makefile (hunk 2 above)
Comment (bsdjhb) [reply lib/csu/aarch64/Makefile:28]: Yes.

File: lib/csu/common/crtend.c (hunk 3 above)
Comment (bsdjhb) [reply lib/csu/common/crtend.c:32]: No, I should perhaps do that though.

File: sys/sys/elf_common.h (hunk 4 above)
Comment (bsdjhb) [reply sys/sys/elf_common.h:749]: Perhaps.

**Pull request merged**